cli/ops/permissions.rs

```diff
@@ -86,7 +86,7 @@ pub fn op_request_permission(
     "env" => Ok(permissions.request_env()),
     "plugin" => Ok(permissions.request_plugin()),
     "hrtime" => Ok(permissions.request_hrtime()),
-    n => Err(type_error(format!("No such permission name: {}", n))),
+    n => Err(other_error(format!("No such permission name: {}", n))),
```

Comment:
http://man7.org/linux/man-pages/man3/errno.3.html

We essentially want to map errors to something like POSIX. In this situation, I think something like EINVAL would be appropriate?

Reply:
Also I think we should remove DenoError.Other - JavaScript already has generic errors TypeError, SyntaxError, etc. Shouldn't we just raise one of those if an appropriate error code isn't found?

Reply:
> http://man7.org/linux/man-pages/man3/errno.3.html
> 
> We essentially want to map errors to something like POSIX. In this situation, I think something like EINVAL would be appropriate?

`ErrorKind.InvalidArgument` sounds fine for me

> Also I think we should remove DenoError.Other - JavaScript already has generic errors TypeError, SyntaxError, etc. Shouldn't we just raise one of those if an appropriate error code isn't found?

That's still gonna be needed as it's part of mechanism of forwarding errors to JS, no problem though to throw `Error` in that case instead of `DenoError`

Reply:
I don't think ES has a defined error class of errors that originate from the platform (e.g. ENOENT) so we might still have to define our own class for that.

For invalid URLs there's https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/URIError and I think we should definitely use that.